packages/fiori/src/ViewSettingsDialog.js

```diff
@@ -0,0 +1,283 @@
+import { fetchI18nBundle, getI18nBundle } from "@ui5/webcomponents-base/dist/i18nBundle.js";
+import { isPhone } from "@ui5/webcomponents-base/dist/Device.js";
+import litRender from "@ui5/webcomponents-base/dist/renderer/LitRenderer.js";
+import UI5Element from "@ui5/webcomponents-base/dist/UI5Element.js";
+import Dialog from "@ui5/webcomponents/Dialog.js";
+import Button from "@ui5/webcomponents/Button.js";
+import Bar from "./Bar.js";
+
+import {
+	VSD_DIALOG_TITLE_SORT,
+	VSD_SUBMIT_BUTTON,
+	VSD_CANCEL_BUTTON,
+	VSD_RESET_BUTTON,
+	VSD_SORT_ORDER,
+	VSD_SORT_BY,
+	VSD_ORDER_ASCENDING,
+	VSD_ORDER_DESCENDING,
+} from "./generated/i18n/i18n-defaults.js";
+
+// Template
+import ViewSettingsDialogTemplate from "./generated/templates/ViewSettingsDialogTemplate.lit.js";
+
+// Styles
+import viewSettingsDialogCSS from "./generated/themes/ViewSettingsDialog.css.js";
+
+const metadata = {
+	tag: "ui5-view-settings-dialog",
+	managedSlots: true,
+	slots: /** @lends  sap.ui.webcomponents.fiori.ViewSettingsDialog.prototype */ {
+		/**
+		 * Defines the <code>Sort By</code> list.
+		 * @type {sap.ui.webcomponents.fiori.List}
+		 * @slot sortBy
+		 * @public
+		 */
+		 "sortBy": {
+			propertyName: "sortBy",
+			type: HTMLElement,
+		},
+	},
+	properties: /** @lends  sap.ui.webcomponents.fiori.ViewSettingsDialog.prototype */ {
+		/**
+		 * Defines the initial sort order.
+		 *
+		 * @type {boolean}
+		 * @defaultvalue false
+		 * @public
+		 */
+		 sortDescending: {
+			type: Boolean,
+		},
+	},
+	events: /** @lends  sap.ui.webcomponents.fiori.ViewSettingsDialog.prototype */ {
+
+		/**
+		 * Fired when OK button is activated.
+		 *
+		 * @event sap.ui.webcomponents.fiori.ViewSettingsDialog#ok-click
+		 * @param {String} sortOrder The current sort order selected.
+		 * @param {String} sortBy The current sort by selected.
+		 * @public
+		 */
+		"ok-click": {
+			detail: {
+				sortOrder: { type: String },
+				sortBy: { type: String },
+			},
```

Comment:
I am not sure if these are the best event names. I would suggest something like submit/confirm and cancel.

Reply:
done